src/auto-evo/mutations/CommonMutationFunctions.cs

```diff
@@ -0,0 +1,237 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+public static class CommonMutationFunctions
+{
+    public static OrganelleDefinition Nucleus = SimulationParameters.Instance.GetOrganelleType("nucleus");
+
+    // These must be defined this way to avoid allocations
+    private static readonly Hex.HexSide[] TraversalOrder1 = [Hex.HexSide.Top];
+    private static readonly Hex.HexSide[] TraversalOrder2 = [Hex.HexSide.Bottom];
+
+    private static readonly Hex.HexSide[] TraversalOrder3 =
+    [
+        Hex.HexSide.TopRight, Hex.HexSide.BottomRight,
+        Hex.HexSide.Top, Hex.HexSide.Bottom, Hex.HexSide.BottomLeft, Hex.HexSide.TopLeft,
+    ];
+
+    private static readonly Hex.HexSide[] TraversalOrder4 =
+    [
+        Hex.HexSide.BottomRight, Hex.HexSide.TopRight,
+        Hex.HexSide.Top, Hex.HexSide.BottomLeft, Hex.HexSide.Bottom, Hex.HexSide.TopLeft,
+    ];
+
+    private static readonly Hex.HexSide[] TraversalOrder5 =
+    [
+        Hex.HexSide.BottomLeft, Hex.HexSide.TopLeft,
+        Hex.HexSide.Bottom, Hex.HexSide.Top, Hex.HexSide.TopRight, Hex.HexSide.BottomRight,
+    ];
+
+    private static readonly Hex.HexSide[] TraversalOrder6 =
+    [
+        Hex.HexSide.TopLeft, Hex.HexSide.BottomLeft,
+        Hex.HexSide.Bottom, Hex.HexSide.Top, Hex.HexSide.BottomRight, Hex.HexSide.TopRight,
+    ];
+
+    private static readonly Hex.HexSide[] TraversalOrder7 =
+    [
+        Hex.HexSide.Top, Hex.HexSide.TopLeft, Hex.HexSide.TopRight,
+        Hex.HexSide.BottomLeft, Hex.HexSide.BottomRight, Hex.HexSide.Bottom,
+    ];
+
+    private static readonly Hex.HexSide[] TraversalOrder8 =
+    [
+        Hex.HexSide.Top, Hex.HexSide.TopRight, Hex.HexSide.TopLeft,
+        Hex.HexSide.BottomRight, Hex.HexSide.BottomLeft, Hex.HexSide.Bottom,
+    ];
+
+    public enum Direction
+    {
+        Front,
+        Neutral,
+        Rear,
+    }
+
+    public static void AddOrganelle(OrganelleDefinition organelle, Direction direction, MicrobeSpecies newSpecies,
+        Random random)
+    {
+        var position = GetRealisticPosition(organelle, newSpecies.Organelles, direction, random);
+
+        // We return early as not being able to add an organelle is not a critical failure
+        if (position == null)
+            return;
+
+        newSpecies.Organelles.Add(position);
+
+        // If the new species is a eukaryote, mark this as such.
+        if (organelle == Nucleus)
+        {
+            newSpecies.IsBacteria = false;
+        }
+    }
+
+    public static OrganelleTemplate? GetRealisticPosition(OrganelleDefinition organelle,
```

Comment:
This method still doesn't have all the upgrades that GetRealisticPosition has on master, including taking in the working memory as parameters to avoid things like line 107.